packages/components/src/Filters/FilterDropdown.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState } from 'react';
+import { Dropdown, DropdownToggle, Level, DropdownProps } from '@patternfly/react-core';
+import FilterInput from './FilterInput.js';
+import './filter-dropdown.scss';
+
+export interface FilterDropdownProps extends Omit<DropdownProps, 'onSelect'> {
+  addFilter: (filterName: string | number, selectedValue: string | number, type: string) => void;
```

Comment:
Most of these props are optional. You can check the original propTypes.

Reply:
You should work around that with conditional access etc. We can't change the API. It will break existing builds.